### DISCUSSION TITLE
[Bug]: Running Linux Native hardware under DietPi

### DISCUSSION THREAD

**Discussion by ShakataGaNai** [meshtastic/firmware#3796] [Troubleshooting]
### Category

Hardware Compatibility

### Hardware

Other

### Firmware Version

2.3.4.ea61808

### Description

At the current time, it is not possible to run the Linux Native Meshtasticd under [DietPi](https://dietpi.com/) v9.2 (A lightweight Debian variant designed for SBC's). While DietPi runs Debian 12 (Bookworm), which is the same core OS as Raspberry Pi OS, there is something in the configuration of DietPi that prevents SPI from working properly with Meshtasticd.

My testing was using a Raspberry Pi 4 B w/ the [Waveshare SX1262 XXXM Lora/GNSS ](https://www.waveshare.com/sx1262-lorawan-hat.htm?sku=24654) HAT. While I2C work as does UART (for the GPS), SPI does not seem to be functioning correctly on a system level, which results in:

```
INFO  | ??:??:?? 10 SX126x init result -2
ERROR | ??:??:?? 10 Failed to find SX1262 radio
```

Running gpioinfo without meshtasticd running looks relatively normal:
[gpioinfo1.txt](https://github.com/meshtastic/firmware/files/14938645/gpioinfo1.txt)

However running gpioinfo whith meshtasticd running shows one major inconsistency, GPIO07 & 08 (spi0 cs1/cs0) are not in use:
[gpioinfo2.txt](https://github.com/meshtastic/firmware/files/14938653/gpioinfo2.txt)

This is the similar output from @jp-bennett who helped me dive into this situation.
```
        line   7:      "GPIO7"   "spi0 CS1"  output   active-low [used]
        line   8:      "GPIO8"   "spi0 CS0"  output   active-low [used]
```

[meshtastic-log.txt](https://github.com/meshtastic/firmware/files/14937422/meshtastic-log.txt)
[config-yaml.txt](https://github.com/meshtastic/firmware/files/14937428/config-yaml.txt)
[config.txt](https://github.com/meshtastic/firmware/files/14937424/config.txt)

**Comment by jp-bennett** [comment]
This isn't officially supported, as the only officially supported target is Raspberry Pi OS Bookworm 64. But we'll use this ticket to try to figure out what went wrong, and hopefully document the steps to make it work.

**Comment by markbirss** [comment]
@ShakataGaNai 

From your meshtastic-log.txt it seems you have defined DIO3_TCXO_VOLTAGE: true but your config.yaml it is not ?

DIO3_TCXO_VOLTAGE: true is not required for the Waveshare SX1262 XXXM Hat only for LoRa Modules with TXCO

**Comment by ShakataGaNai** [comment]
Oh this got converted do a discussion rather than closed. Meshtastic worked under DietPi (not officially speaking, but it should work just fine). After much help from JP, the issue was bad hardware, not bad software.
